src/sentry/api/endpoints/group_integration_details.py

```diff
@@ -5,10 +5,42 @@
 from rest_framework.response import Response
 
 from sentry.api.bases import GroupEndpoint
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.integration import IntegrationIssueSerializer
+from sentry.integrations.base import IntegrationFeatures
 from sentry.models import ExternalIssue, GroupLink, OrganizationIntegration
 
 
 class GroupIntegrationDetails(GroupEndpoint):
+    def get(self, request, group, integration_id):
+        # Keep link/create separate since create will likely require
+        # many external API calls that aren't necessary if the user is
+        # just linking
+        action = request.GET.get('action')
+        if action not in {'link', 'create'}:
+            return Response({'detail': 'Action is required and should be either link or create'})
+
+        organization_id = group.project.organization_id
+        try:
+            # check org permissions
+            # TODO(jess): should this eventually check ProjectIntegration?
+            integration = OrganizationIntegration.objects.filter(
+                integration_id=integration_id,
+                organization_id=organization_id,
+            ).select_related('integration').get().integration
+        except OrganizationIntegration.DoesNotExist:
+            return Response(status=404)
+
+        provider = integration.get_provider()
+        if IntegrationFeatures.ISSUE_SYNC not in provider.features:
```

Comment:
This solution seems ok to me. For the features, if you really wanted to you could have used a decorator. Or a simpler middle-ground is probably just creating a method on the model that takes in a given feature and returns either true or false if it is on that integration or not. Honestly this is fine though.

Reply:
oh yeah i like the idea of a method like `has_feature`. lemme try doing that